timemachine/cpp/src/math_utils.cuh

```diff
@@ -1,9 +1,11 @@
 #pragma once
 
-__device__ __host__ double sign(double a) { return (a > 0) - (a < 0); }
+int __forceinline__ ceil_divide(int x, int y) { return (x + y - 1) / y; }
```

Comment:
I would like this to be more widely used, but for now I just used it for the code that relate to these changes.

Reply:
Would it be okay if we kept the old sign code as well? Not sure why it was deleted.

Reply:
Its only used by the GBSA potential which isn't currently compiled. But makes sense to keep it if we don't also remove the GBSA potential.

Reply:
Okay, these are really just jvp rules for autodiff. Let's keep them for now since they may be useful later on?